src/web/parserUtils.ts

```diff
@@ -78,6 +78,17 @@ function addSubstringAsTextNode(root: HTMLElement, text: string, startIndex: num
   }
 }
 
+function ungroupRanges(ranges: MarkdownRange[]): MarkdownRange[] {
+  const ungroupedRanges: MarkdownRange[] = [];
+  ranges.forEach((range) => {
+    if (!range.depth) {
+      ungroupedRanges.push(range);
+    }
+    Array.from({length: range.depth!}).forEach(() => ungroupedRanges.push(range));
```

Comment:
Since we are doing it before parsing, there is no need to do so, the main loop of the parser is not using the info about depth

Reply:
I understand, it's just that it could be confusing to still see `depth` in there, like:
```
[
  {style: "blockquote", start: 0, end: 0, depth: 3},
  {style: "blockquote", start: 0, end: 0, depth: 3},
  {style: "blockquote", start: 0, end: 0, depth: 3},
]
```

Reply:
You're right, I'll add removing it